src/lib.rs

```diff
@@ -320,6 +367,104 @@ mod tests {
         assert_eq!(res, [3, 2, 1]);
     }
 
+    // #[test]
+    // fn popping_then_pushing_out_of_scope() {
+    //     // should fail to compile !!!
```

Comment:
Rustdoc has a thing that tells it the test should fail to compile (`compile_fail`). Please replace this with a doctest that tries to `pop()` from a `CircularQueue<String>` with that `compile_fail` "attribute".